core/server/session_logs.go

```diff
@@ -114,35 +126,191 @@ func (cs *coreServer) GetSessionLogs(ctx context.Context, msg *pb.GetSessionLogs
 		},
 	)
 	if err != nil {
-		return nil, err
+		return &pb.GetSessionLogsResponse{Error: err.Error()}, err
+	}
+
+	var (
+		logEntries []*pb.LogEntry
+		firstToken string
+	)
+
+	if msg.GetLogSourceFilter() == "" || msg.GetLogSourceFilter() == logger.SessionLogSource {
```

Comment:
❓  Can we assign `msg.GetLogSourceFilter()` to a constant for these checks in order not to make this call twice?

Reply:
I would have done that. Thank you Olga! Will fix it.

Reply:
OK, `const x = msg.GetLogSourceFilter()` didn't work. 
This kind of getter would get optimized nicely by Go. I think we don't have to worry much about it.

Reply:
Sorry, wrote const but meant smth. like using a regular var `x := msg.GetLogSourceFilter()` (well, it was late at night while I was reviewing it 😅  ). But it's of very minor concern.